internal/kafka/internal/services/clusters.go

```diff
@@ -285,7 +284,7 @@ func (c clusterService) FindCluster(criteria FindClusterCriteria) (*api.Cluster,
 		if errors.Is(err, gorm.ErrRecordNotFound) {
 			return nil, nil
 		}
-		return nil, apiErrors.NewWithCause(apiErrors.ErrorGeneral, err, "failed to find cluster with criteria")
+		return nil, err
```

Comment:
can return the database error as is since it's the only place that returns an error in this method. The caller of this method can wrap this error instead.